src/pip/_internal/baseparser.py

```diff
@@ -192,7 +192,14 @@ def _update_defaults(self, defaults):
                 continue
 
             if option.action in ('store_true', 'store_false', 'count'):
-                val = strtobool(val)
+                try:
+                    val = strtobool(val)
+                except ValueError:
+                    self.error(
+                        self.invalid_config_error_message(
```

Comment:
I'd make this a variable:

```
msg = invalid.....
self.error(msg)
```

Reply:
Lol, earlier it was a variable. With new changes I moved it inside. Yeah, I will do that.